packages/lib/modules/pool/pool-tokens.utils.ts

```diff
@@ -290,3 +317,52 @@ export function getPriceRateForToken(token: ApiToken, pool: Pool) {
   return pool.poolTokens.find(poolToken => poolToken.underlyingToken?.address === token.address)
     ?.priceRate
 }
+
+/* Given a token (wrapped or underlying):
+  - If wrapped token: returns a function that returns the wrapped token with its corresponding underlying token
+  - If underlying token: returns a function that returns the underlying token with its corresponding wrapped token
+
+  Sorted by:
+  - token with more balance first
+  - underlying token first
+*/
+export function getWrappedAndUnderlyingTokenFn(
+  token: ApiToken,
+  pool: Pool,
+  balanceFor: BalanceForFn
+): () => [ApiToken, ApiToken] | void {
+  if (shouldUseUnderlyingToken(token, pool)) {
+    return () => {
+      // TODO: Review if we should exclude some info to avoid wrapped being shouldUseUnderlyingToken
+      const underlyingToken = { ...token, ...token.underlyingToken, wrappedToken: token }
+      const wrappedToken = token
+
+      return [underlyingToken, wrappedToken]
+    }
+  }
+
+  if (token.wrappedToken) {
+    return () => {
+      const underlyingToken = token
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const wrappedToken = token.wrappedToken!
```

Comment:
move `const wrappedToken = token.wrappedToken` to below the `if` statement and you then don't need the null assertion

Reply:
Great!